### PR TITLE
[chore] set default transport defaults on NewDefaultClientConfig

### PR DESCRIPTION
**Pull request by atoulme**
#### Description
Set the defaults on `NewDefaultClientConfig`. This change is a no-op since those defaults are set if the value is not set.

Add documentation to all fields as well.

#### Link to tracking issue
#9478

### CODE REVIEW

File: config/confighttp/confighttp.go [hunk 1]
```diff
@@ -73,20 +76,20 @@ type ClientConfig struct {
...
 
 	// MaxConnsPerHost limits the total number of connections per host, including connections in the dialing,
 	// active, and idle states.
-	// There's an already set value, and we want to override it only if an explicit value provided
+	// By default, it is set to http.DefaultTransport.MaxConnsPerHost (0, unlimited).
 	MaxConnsPerHost *int `mapstructure:"max_conns_per_host"`
 
 	// IdleConnTimeout is the maximum amount of time a connection will remain open before closing itself.
-	// There's an already set value, and we want to override it only if an explicit value provided
+	// By default, it is set to http.DefaultTransport.IdleConnTimeout (90s)
```
Comment (mx-psi) [inline config/confighttp/confighttp.go:92]: I would remove the value in braces, we can add a [Docs link](https://go.dev/doc/comment#doclinks) instead so we don't have to keep the value updated

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
There's some errors in the `httpforwarderextension`, could you take a look at those and see if we can fix them before merging this PR?

**Review by TylerHelmuth — APPROVED** [review]

**Comment by atoulme** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34477 is open to unblock.

**Comment by mx-psi** [comment]
```
=== FAIL: . TestConfig/elasticsearch/metric (re-run 1) (0.01s)
    config_test.go:319: 
        	Error Trace:	/tmp/opentelemetry-collector-contrib/exporter/elasticsearchexporter/config_test.go:319
```

**Pull request merged**